import.sh

```diff
@@ -150,18 +153,33 @@ import() {
 
 	# At this point, the file has been saved to the cache so
 	# either source it or print it.
-	if [ -z "${print-}" ]; then
+	if [ "$action" = "source" ]; then
 		[ -n "${IMPORT_DEBUG-}" ] && echo "import: sourcing '$cache_path'" >&2
 		local __import_parent_location="${__import_location-}"
 		__import_location="$(cat "$cache/locations/$url_path")" || return
 		. "$cache_path" || return
 		__import_location="$__import_parent_location"
-	else
+	elif [ "$action" = "print" ]; then
 		[ -n "${IMPORT_DEBUG-}" ] && echo "import: printing '$cache_path'" >&2
 		echo "$cache_path"
+	else
+		echo "import_internal called with unknown action '$action'. Please use import or import_file." >&2
+		return 2
 	fi
 }
 
+# Download & cache and echo the local path to stdout
+import_file() {
+	import_internal print "$@"
```

Comment:
Wouldn't it be easier to leave `import` as-is and just do:

```suggestion
	print=1 import "$@"
```

Reply:
You're right.